app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataService.kt

```diff
@@ -127,6 +135,18 @@ class DefaultHistoryMetadataService(
             storage.deleteHistoryMetadataOlderThan(olderThan)
         }
     }
+
+    override fun removeMetadata(keys: List<HistoryMetadataKey>) {
+        scope.launch(IO) {
+            keys.forEach { key ->
+                try {
+                    storage.deleteHistoryMetadata(key)
+                } catch (e: Exception) {
+                    Logger.error("Failed to remove history metadata: $e")
+                }
```

Comment:
Not sure if this is necessary here (nothing was mentioned in the kdocs for `.deleteHistoryMetadata`) but I figured safer than sorry?

Reply:
@eliserichards @grigoryk just opened a PR to simplify this so you can just call `storage.deleteHistoryMetadata(searchTerms)`. We will publish today and upgrade Fenix. This should simplify this quite a bit and you don't need to add a method to the service.